### PR TITLE
E2E test for ClusterClass feature and CC docker templates

### LINKED ISSUE #538: Add unit/Integration test for cluster class support in BYOH
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Related to #487 

**Anything else you would like to add:**
[Miscellaneous information that will assist in implementing the feature.]

**Environment:**

- Cluster-api-provider-bringyourownhost version: 
- Kubernetes version: (use `kubectl version --short`):
- OS (e.g. from `/etc/os-release`):

### LINKED ISSUE #539: [E2E] e2e test to verify cluster class support integration with BYOH
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
Related to #487 

**Anything else you would like to add:**
[Miscellaneous information that will assist in implementing the feature.]

**Environment:**

- Cluster-api-provider-bringyourownhost version: 
- Kubernetes version: (use `kubectl version --short`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by shivi28**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #538 
Fixes #539 

**Additional information**

**Special notes for your reviewer**

### CODE REVIEW

File: test/e2e/config/provider.yaml [hunk 1]
```diff
@@ -75,6 +75,8 @@ providers:
         files:
           # Add a cluster template
           - sourcePath: "../data/infrastructure-provider-byoh/v1beta1/templates/e2e/cluster-template.yaml"
+          - sourcePath: "../data/infrastructure-provider-byoh/v1beta1/templates/e2e/clusterclass-quickstart.yaml"
```
Comment (shamsher31) [inline test/e2e/config/provider.yaml:78]: Not sure if this is the right way to provide a path, Is there a way we can provide `basepath` in this template?

File: test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/e2e/cluster-template-topology.yaml [hunk 2]
```diff
@@ -81,7 +81,142 @@ spec:
...
+        files:
+        - owner: root:root
+          path: /etc/kubernetes/manifests/kube-vip.yaml
+        initConfiguration:
+          nodeRegistration:
+            criSocket: /var/run/containerd/containerd.sock
+            ignorePreflightErrors:
+              - Swap
+              - DirAvailable--etc-kubernetes-manifests
+              - FileAvailable--etc-kubernetes-kubelet.conf
```
Comment (shamsher31) [inline test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/e2e/cluster-template-topology.yaml:117]: Just for our reference:
@mayur-tolexo I think we can pass this ignore errors for `containerd` check-raised by CDS?

File: test/e2e/config/provider.yaml (hunk 1 above)
Comment (shivi28) [reply test/e2e/config/provider.yaml:78]: For uniformity keeping it same as other providers, referred [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/e2e_conf.yaml#L209) .

File: test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/e2e/cluster-template-topology.yaml [hunk 3]
```diff
@@ -11,22 +11,22 @@ spec:
...
       cidrBlocks:
-      - "192.168.0.0/16"
-    serviceDomain: "cluster.local"
+      - 192.168.0.0/16
+    serviceDomain: cluster.local
   topology:
-    class: ${CLUSTER_CLASS_NAME}
-    version: ${KUBERNETES_VERSION}
+    class: quickstart
+    version: v1.23.5
```
[the remark this answers is not in the document]
Comment (shivi28) [reply test/e2e/data/infrastructure-provider-byoh/v1beta1/templates/e2e/cluster-template-topology.yaml]: done

**Review by anusha94 — DISMISSED** [review]

**Review by shamsher31 — APPROVED** [review]
LGTM 👍

**Review by anusha94 — APPROVED** [review]

**Pull request merged**